### ISSUE TITLE
Exception in Weight Learning with 0 Grounding Rules

### ISSUE THREAD

**Issue by eriq-augustine** [eriq-augustine/psl#63]
Looks like some rules (logical?) with no groundings are causing exceptions in weight learning.

Sabina has seen this and I think I may have seen it before.

---
```
Exception in thread "main" java.lang.IllegalAccessError: tried to access method com.google.common.collect.Iterators.emptyIterator()Lcom/google/common/collect/UnmodifiableIterator; from class de.mathnbits.util.KeyedRetrievalSet
    at de.mathnbits.util.KeyedRetrievalSet.iterator(KeyedRetrievalSet.java:87)
    at de.mathnbits.util.KeyedRetrievalSet$2.iterator(KeyedRetrievalSet.java:104)
    at org.linqs.psl.application.learning.weight.maxlikelihood.MaxLikelihoodMPE.computeObservedIncomp(MaxLikelihoodMPE.java:75)
    at org.linqs.psl.application.learning.weight.maxlikelihood.VotedPerceptron.doLearn(VotedPerceptron.java:250)
    at org.linqs.psl.application.learning.weight.WeightLearningApplication.learn(WeightLearningApplication.java:120)
    at org.linqs.psl.application.learning.weight.WeightLearningApplication$learn.call(Unknown Source)
    at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
    at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:110)
    at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:114)
```

**Comment by eriq-augustine** [comment]
Confirmed: https://github.com/eriq-augustine/psl/tree/eriq-bug-zeroground
Looks like de.mathnbits is returning an Iterable where we are not allowed to get it's Iterator because it is empty.

The proper way to fix this would be to not use de.mathnbits.util.KeyedRetrievalSet.
Probably a good decision since I cannot find ANY documentation on it.

**Comment by eriq-augustine** [comment]
Fixed in 5e6b51a324493ffb7a5bb71c96437e8e049914e4 by switching libraries away from de.mathnbts. #64 
Now using org.apache.commons.collections4.MultiValuedMap instead of de.mathnbits.util.KeyedRetrievalSet
